src/HttpLayer.hs

```diff
@@ -40,41 +42,62 @@ emptyHTTPNetworkLayer = HTTPNetworkLayer
 ------------------------------------------------------------
 
 -- | General api request function
-apiRequest :: Config -> Text -> Request
-apiRequest Config{..} u =
-    setRequestPath (encodeUtf8 path) $
-    addRequestHeader "Content-Type" "application/json" $
-    setRequestBasicAuth
-        (encodeUtf8 cfgEmail <> "/token")
-        (encodeUtf8 cfgToken) $
-    parseRequest_ (toString (cfgZendesk <> path))
+apiRequest
+    :: Config
+    -> Text
+    -> Either String Request
+apiRequest Config{..} u = mapLeft show $ catches buildRequest handlerList
```

Comment:
I don't think we need an `Either` here. All our URLs are "internal" and will fail if incorrect anyway.
I would otherwise consider this, but it seems to be difficult for you right now.

Reply:
There are multiple external functions that get called there and they can fail with an `HttpException` or with a `JSONException`. We shouldn't be hoping it will always be a happy case.

Reply:
Where in does `apiRequest` it says that?

Reply:
There is a `handlerList` in the body of the function providing these two types of exceptions.

Reply:
Maybe I'm getting something wrong, but there is this function call: `catches buildRequest handlerList`. In case `buildRequest` results in an exception given in the `handlerList`, it gets handled by a corresponding handler.